src/plugins/vis_type_timeseries/public/application/components/aggs/top_hit.js

```diff
@@ -120,7 +120,7 @@ const TopHitAggUi = (props) => {
   const handleSelectChange = createSelectHandler(handleChange);
   const handleTextChange = createTextHandler(handleChange);
   const fieldsSelector = getIndexPatternKey(indexPattern);
-  const field = fields[fieldsSelector].find((f) => f.name === model.field);
+  const field = fields?.[fieldsSelector]?.find((f) => f.name === model.field) ?? {};
```

Comment:
The only consumer of `field` is `getAggWithOptions` which already has it initialized to `{}`. Perhaps we could avoid the `?? {}` here and let the `undefined` handler do its work?

Reply:
oh yes good point